lib/internal/Magento/Framework/ObjectManager/Config/Config.php

```diff
@@ -156,6 +156,7 @@ public function isShared($type)
     public function getInstanceType($instanceName)
     {
         while (isset($this->_virtualTypes[$instanceName])) {
+            if ($instanceName == $this->_virtualTypes[$instanceName]) break;
```

Comment:
I believe it's better to throw exception here

Reply:
Do you have a preference between:
ConfigurationMismatchException, and
IntegrationException ?

Reply:
this exception should not be user facing, so i believe \Exception is ok here

Reply:
Following the anti-recursion pattern in the following function, I modified the code to throw a `\LogicException` instead of exiting the infinite loop and returning the self-referencing instance.